autorest/models/parameter.py

```diff
@@ -51,8 +50,8 @@ def __init__(
     ):
         self.yaml_data = yaml_data
         self.schema = schema
-        self.name = name
-        self.serialized_name = get_parameter_name(name)
+        self.rest_api_name = rest_api_name
+        self.serialized_name = yaml_data['language']['python']['name']
```

Comment:
Please add "serialized_name" in the __init__, and pull it from the yaml in from_yaml instead

Reply:
switched